src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1027,6 +1028,7 @@ def send_raw_request(cli_ctx, method, url, headers=None, uri_parameters=None,  #
 
     # https://requests.readthedocs.io/en/latest/user/advanced/#prepared-requests
     s = Session()
+    s.mount(url, SSLContextAdapter())
```

Comment:
Applying `SSLContextAdapter` in `send_raw_request()` will only affect `az rest`. All other commands that use Azure Python SDK will not benefit from this change. MSAL is not affected either (https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/685).